src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -127,58 +148,55 @@ class HttpGoogleCloudStorageDAO(
             .setCondition(new Condition().setAge(0))
           val lifecycle = new Lifecycle().setRule(List(deleteEverythingRule))
           val fetcher = buckets.get(bucketName)
-          val bucket = retry(fetcher.execute,when500)
-          bucket.setLifecycle(lifecycle)
+          val bucketFuture = retry(when500)(() => Future { fetcher.execute })
```

Comment:
ive noticed we seem to do a lot of retry(when500)(() => Future { , should we generalize that?